packages/ui/src/components/filters/filters-bar/actions/views.tsx

```diff
@@ -0,0 +1,241 @@
+import { FC, useEffect, useState } from 'react'
+
+import {
+  AlertDialog,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+  Button,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  Icon,
+  Input,
+  MessageTheme
+} from '@/components'
+
+import { SavedView } from '../../types'
+import { UseFiltersReturn } from '../../use-filters'
+
+interface ViewsProps {
+  currentView: SavedView | null
+  savedViews: SavedView[]
+  saveView: UseFiltersReturn['saveView']
+  updateView: UseFiltersReturn['updateView']
+  deleteView: UseFiltersReturn['deleteView']
+  renameView: UseFiltersReturn['renameView']
+  hasChanges: boolean
+}
+
+const Views: FC<ViewsProps> = ({
+  currentView,
+  savedViews,
+  saveView,
+  updateView,
+  deleteView,
+  renameView,
+  hasChanges
+}) => {
+  const [isNewViewDialogOpen, setIsNewViewDialogOpen] = useState(false)
+  const [isRenameDialogOpen, setIsRenameDialogOpen] = useState(false)
+  const [newViewName, setNewViewName] = useState('')
+  const [nameError, setNameError] = useState('')
+
+  /**
+   * Checks if a view name already exists
+   * @param name - Name to check
+   * @param excludeId - Optional ID to exclude from check (used when renaming)
+   */
+  const checkNameExists = (name: string, excludeId?: string) => {
+    return savedViews.some(view => view.name.toLowerCase() === name.toLowerCase() && view.id !== excludeId)
+  }
+
+  /**
+   * Manages view name state when rename dialog opens/closes
+   * - Sets current view name when rename dialog opens
+   * - Clears name when dialog closes
+   */
+  useEffect(() => {
+    if (!currentView) return
+
+    if (isRenameDialogOpen && currentView) {
+      setNewViewName(currentView.name)
+    } else if (!isRenameDialogOpen) {
+      setNewViewName('')
+    }
+  }, [isRenameDialogOpen, currentView])
+
+  /**
+   * Handles saving a new view
+   * - Validates name is not empty
+   * - Checks for duplicate names
+   * - Saves view and closes dialog
+   */
+  const handleSaveNewView = () => {
+    if (!newViewName.trim()) return
+
+    if (checkNameExists(newViewName)) {
+      setNameError('A view with this name already exists.')
+      return
+    }
+
+    saveView(newViewName)
+    setNewViewName('')
+    setNameError('')
+    setIsNewViewDialogOpen(false)
+  }
+
+  /**
+   * Handles renaming existing view
+   * - Validates current view exists
+   * - Checks for duplicate names
+   * - Updates view name and closes dialog
+   */
+  const handleRename = () => {
```

Comment:
same thing for all callbacks.. there should be no business logic in the `ui` package components and views. these react components should be presentation-only, and take callbacks as props for all behaviour

Reply:
@abhinavrastogi-harness I’ve revised the approaches, so now only the local logic necessary for this component remains. It handles its own state, while all the business logic for filtering and validation has been moved to a hook called [use-view-management](https://github.com/harness/canary/pull/513/files#diff-c58401d04ae9a913353d34aeabc42330af54d16582e87e43325c039034e9c35bR43), located at `views/repo`.